### DISCUSSION TITLE
allow multiple AUR packages

### DISCUSSION THREAD

**Discussion by k0kubun** [xremap/aur-autoupdater#72] [Ideas]
Originally filed by @rikutons:

---

I'm using multiple desktop env on 1 computer.
But `xremap-x11-bin` and `xremap-wlroots-bin` are conflicting at `/usr/bin/xremap` so I can't switch them easily.

`paru -S xremap-x11-bin` then `paru -S xremap-wlroots-bin` makes
```
error: failed to commit transaction (conflicting files)
xremap-wlroots-bin: /usr/bin/xremap exists in filesystem (owned by xremap-x11-bin)
Errors occurred, no packages were upgraded.
```

For example in case x11, can you make `xremap-x11-bin`'s binary name to `/usr/bin/xremap-x11` and create symlink to `/usr/bin/xremap` if `/usr/bin/xremap` is empty, or always override it to allow multiple xremap's packeges?

**Comment by k0kubun** [comment]
I'm open to adopting that feature. Would you be interested in working on a pull request for it? You might be able to achieve it by modifying `assets/xremap-*-bin/PKGBUILD.tmpl`.

**Comment by szdytom** [comment]
I suggest we move the binaries to `/usr/bin/xremap-{env}` and make `/usr/bin/xremap` a shell script that checks the existence of each binary and invokes the first found. The shell script can be an separate package and the binary packages marks it a an dependence.

**Comment by k0kubun** [comment]
That sounds like a good idea. Could you file a PR that implements it?

**Comment by szdytom** [comment]
@k0kubun Sorry for the long delay—over a year, I know. I've recently switched to KDE, and now I actually need this feature myself. I'm finally working on a PR to implement it.

However, I noticed that the xremap-kde-bin and xremap-niri-bin packages are not managed by the same templating script (they seem to be maintained by @Stitchless and @RimuruW (Qingxu on AUR) respectively). Have you reached out to them about aligning with this change?

**Comment by szdytom** [comment]
I've [commented on AUR page of xremap-kde-bin](https://aur.archlinux.org/packages/xremap-kde-bin#comment-1066821). I'll wait a few days for Stitchless to response.

**Reply by k0kubun** [reply]
Apparently they have responded and added me as a co-maintainer. Pushed https://github.com/xremap/aur-autoupdater/pull/94.

**Comment by szdytom** [comment]
I've put together [a draft wrapper script](https://paste.ubuntu.com/p/kNSvBPc9kz/) that attempts to auto-detect the current graphical session. Since I don't have access to every supported environment for testing, I'd really appreciate it if you could take a look and give it a try when you have a moment. Thanks! @k0kubun @RimuruW

**Reply by k0kubun** [reply]
right now I'm away from my machine, but the patch looks good :+1:
